packages/database/src/migrations/20210629030241-WeeklyCasesDashboardForFijiDemoFacility-modifies-data.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+import { insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const DISEASE_DATA_ELEMENTS = {
+  AFR: {
+    dataElement: 'PSSS_AFR_Cases',
+    name: 'Acute Fever and Rash',
+    color: '#F0965BFF', // orange
+  },
+  DIA: {
+    dataElement: 'PSSS_DIA_Cases',
+    name: 'Diarrhoea',
+    color: '#81DEE4FF', // aqua
+  },
+  ILI: {
+    dataElement: 'PSSS_ILI_Cases',
+    name: 'Influenza Like Illness',
+    color: '#4DA347FF', // green
+  },
+  PF: {
+    dataElement: 'PSSS_PF_Cases',
+    name: 'Prolonged Fever',
+    color: '#1C49A7FF', // blue
+  },
+  DLI: {
+    dataElement: 'PSSS_DLI_Cases',
+    name: 'Dengue Like Illness',
+    color: '#8455F6', // purple
+  },
+};
+
+const getDashboardReportId = diseaseCode =>
+  `PSSS_FJ_${diseaseCode}_Weekly_Case_Trend_Graph_Facility`;
+
+const dashboardGroupCode = 'FJ_PSSS_Syndromic_Surveillance_National_Data_Facility_Public';
+
+const getDashboardReport = (id, diseaseName, dataElementCode, color) => ({
+  id,
+  dataBuilder: 'analyticsPerPeriod',
+  dataBuilderConfig: {
+    dataElementCode,
+    aggregations: [
+      {
+        type: 'FINAL_EACH_WEEK_FILL_EMPTY_WEEKS',
+        config: {
+          fillEmptyPeriodsWith: null,
+        },
+      },
+    ],
+    entityAggregation: {
+      dataSourceEntityType: 'facility',
+    },
+  },
+  viewJson: {
+    name: `${diseaseName} Weekly Case Number Trend Graph`,
+    type: 'chart',
+    chartType: 'line',
+    chartConfig: {
+      value: {
+        color,
+      },
+    },
+    defaultTimePeriod: {
+      end: {
+        unit: 'week',
+        offset: 0,
+      },
+      start: {
+        unit: 'week',
+        offset: -52,
+      },
+    },
+    presentationOptions: {
+      periodTickFormat: '[W]w',
+    },
+    periodGranularity: 'week',
+  },
+});
+
+exports.up = async function (db) {
+  const dashboardReportIds = [];
+  for (const [diseaseCode, { dataElement, name, color }] of Object.entries(DISEASE_DATA_ELEMENTS)) {
+    const dashboardReportId = getDashboardReportId(diseaseCode);
+    dashboardReportIds.push(dashboardReportId);
+
+    const dashboardReport = getDashboardReport(dashboardReportId, name, dataElement, color);
+    await insertObject(db, 'dashboardReport', dashboardReport);
+  }
```

Comment:
@rohan-bes Yes we share the same first thought just to copy the existing Palau visuals to Fiji, but they are using different dataElements (see [commit](https://github.com/beyondessential/tupaia/pull/2753/commits/9f6af95aaf94ab9cabe600013d6ea694c210369b)). One is using indicators, the new one is using new dataElements.

I like your idea to remove `FJ` from the id though, in case we will use them in the future👍

Reply:
Is there a reason we can't use the same indicators in this dashboard as well? If querying at the site level, then the value should be the same right?

Reply:
No that's right we can use the same visuals for dashboards. We need to update the previous visuals ids across both legacy and new tables.